README.md

```diff
@@ -21,7 +26,66 @@ containing ascii characters):
 assert(scopeUtils.validScope("..."));
 ```
 
-### Checking Scope Sets
+### Scope Expressions
+
+These are the "new-style" way of dealing with scopes and allow for greater
+flexibility than the old style. The expressions are objects with a single key --
+either `AnyOf` or `AllOf`. The value must be an array and the elements of the
+array must be either expressions or a [valid scope](#valid-scopes).
+
+A scope expression can be evaluated against an array of scopes to determine if the
+scope expression is "satisfied" by the array of scopes. Satisfaction in this context
+means that enough scopes in the expression are either in the array directly or are
+prefixed by a scope in the array that ends in `*` to allow the `AnyOf` and `AllOf`
+clauses to be true.
+
+The semantics of these clauses are:
+
+**AllOf:** All sub-expressions and scopes must be satisfied
+
+**AnyOf:** At least one sub-expression or scope must be satisfied
+
+#### Scope Expression Examples
+
+
+```js
+// Evaluates to true
+scopeUtils.satisfiesExpression(
+  [
+    'abc*',
+  ],
+  {
+    AnyOf: ['abcd'],
+  }
+)
+
+// Evaluates to false
+scopeUtils.satisfiesExpression(
+  [
+    'abc*',
+  ],
+  {
+    AnyOf: ['def'],
+  }
+)
+
+// Evaluates to true
+scopeUtils.satisfiesExpression(
+  [
+    'abc*',
+  ],
+  {
+    AnyOf: [
+      {AllOf: ['abcdef']},
+      'def',
+    ]
+  }
+)
+```
+
+### Local Authorization (Old Style)
```

Comment:
`### Nested Arrays (Old Style)`

I think it makes sense to add a paragaraph here about equivalence, too, since that makes it clear these do the same job, just less expressively [1].

"The old method of checking satisfaction involved nested arrays confusing referred to as "scopesets".  The scopesets `[['a', 'b'], ['c']]` is equivalent to the scope expression `{AnyOf: [{AllOf: ['a', 'b']}, {AllOf: ['c']}]}` or more concisely `{AnyOf: [{AllOf: ['a', 'b']}, 'c']}`."

[1] OK, not strictly -- any scope expression can has an equivalent scopesets in disjunctive normal form, but that form might be huge.